input.cpp

```diff
@@ -2752,6 +2766,10 @@ int input_test(int getchar)
 							input[n].vid = id.vendor;
 							input[n].pid = id.product;
 
+							auto result = devnums.find(device_key_string(id.vendor, id.product));
+							if (result != devnums.end())
+								input[n].num = result->second;
```

Comment:
With `num` being set here, it's now possible for the loop in `input_cb()` to assign an already-used `num` value to an additional device, per this assignment at line 1707:

```
			if (!found)
			{
				input[dev].num = num;
				break;
			}
```
So in addition to the changes I've made already, that loop will need to avoid assigning `num` values that are already in use by other input devices.

Reply:
Actually, the longer I look at that preexisting nested `for` loop, the more I realize that what it's doing doesn't make much sense.  I think it's already assigning `num` values incorrectly, even without my change.

Reply:
I rewrote the loop to do the right thing, and I've committed that addition to this pending PR.

That fixed the last issue I was seeing.  (Sometimes when I reconnected one device out of my three [keyboard, gamepad1, gamepad2], it wouldn't work as expected (because it was assigned the wrong `num`), or its input would stutter/lag (because another unrelated input device was already using the same `num`).